src/main/kotlin/com/openlattice/search/renderers/CAREIssueAlertEmailRenderer.kt

```diff
@@ -0,0 +1,186 @@
+private const val STAFF_ENTITY_SET_ID_METADATA = "staffEntitySetId"
+private const val TIME_ZONE_METADATA = "timezone"
+private const val ASSIGNED_TO_ENTITY_SET_ID_METADATA = "assignedToEntitySetId"
+private const val REPORTED_ENTITY_SET_ID_METADATA = "reportedEntitySetId"
+
+/*
+*
+* Search metadata is expected to contain the following fields:
+* {
+*   "personEntitySetId": <UUID>,
+*   "staffEntitySetId": <UUID>,
+*   "assignedToEntitySetID": <UUID>,
+*   "reportedEntitySetID": <UUID>,
+*   "timezone": <TimeZones>,
+*   "alertName": <String>,
+* }
+*
+*/
+
+class CAREIssueAlertEmailRenderer {
+
+    companion object {
+
+        private fun isMatchByEntitySetId(neighbor :NeighborEntityDetails, entitySetId: UUID, associationEntitySetId: Optional<UUID>): Boolean {
+            val isMatchingEntitySetId = neighbor.neighborEntitySet.isPresent && neighbor.neighborEntitySet.get().id == entitySetId
+            return when (associationEntitySetId.isPresent) {
+                (neighbor.associationEntitySet.id == associationEntitySetId.get()) && isMatchingEntitySetId -> isMatchingEntitySetId
+                else -> isMatchingEntitySetId
+            }
+        }
+
+        private fun getCombinedNeighbors(
+                neighbors: List<NeighborEntityDetails>, entitySetId: UUID, associationEntitySetId: Optional<UUID>
+        ): Map<FullQualifiedName, Set<Any>> {
+            val entity = mutableMapOf<FullQualifiedName, MutableSet<Any>>()
+            neighbors.filter { isMatchByEntitySetId(it, entitySetId, associationEntitySetId) }.forEach {
+                it.neighborDetails.get().forEach { (fqn, values) ->
+                    entity.getOrPut(fqn) { mutableSetOf() }.addAll(values)
+                }
+            }
+
+            return entity
+        }
+
+        private fun getPersonDetails(
+                neighbors: List<NeighborEntityDetails>, personEntitySetId: UUID
+        ): Map<String, Any> {
+            val combinedEntity = getCombinedNeighbors(neighbors, personEntitySetId, Optional.empty())
+
+            val tags = mutableMapOf<String, String>()
+
+            val firstName = (combinedEntity[FIRST_NAME_FQN] ?: emptySet()).joinToString("/")
+            val lastName = (combinedEntity[LAST_NAME_FQN] ?: emptySet()).joinToString("/")
+            val middleName = (combinedEntity[MIDDLE_NAME_FQN] ?: emptySet()).joinToString("/")
+            tags["formattedName"] = "$lastName, $firstName $middleName"
+
+            return tags
+        }
+
+        private fun getReporterDetails(
+                neighbors: List<NeighborEntityDetails>, staffEntitySetId: UUID, reporterEntitySetId: UUID
+        ): Map<String, Any> {
+            val combinedEntity = getCombinedNeighbors(neighbors, staffEntitySetId, Optional.of(reporterEntitySetId))
+
+            val tags = mutableMapOf<String, String>()
+
+            tags["reporter"] = (combinedEntity[PERSON_ID_FQN] ?: emptySet()).joinToString(", ")
+
+            return tags
```

Comment:
Also here I think returning a pair is simpler but v.minor. Would pref sizing that tags map tho if no Pair